### PR TITLE
Tx: Improve Code Documentation

### PR DESCRIPTION
**Pull request by holgerd77**
This PR improves the tx library code documentation based on various recent discussions along the publication of the new EIP-1559/London compatible tx library release.

### CODE REVIEW

File: packages/tx/src/baseTransaction.ts [hunk 1]
```diff
@@ -156,11 +160,10 @@ export abstract class BaseTransaction<TransactionObject> {
...
 
-  /**
-   * Returns the serialized unsigned tx (hashed or raw), which is used to sign the transaction.
-   *
-   * @param hashMessage - Return hashed message if set to true (default: true)
-   */
+  // Returns the serialized unsigned tx (hashed or raw), which is used to sign the transaction.
+  //
+  // Note: do not use code docs here since VS Studio is then not able to detect the
+  // comments from the inherited methods
```
Comment (ryanio) [inline packages/tx/src/baseTransaction.ts:166]: is this the reason for the change here from `/**` to `//` style comments?

File: packages/tx/src/eip1559Transaction.ts [hunk 2]
```diff
@@ -234,15 +260,30 @@ export default class FeeMarketEIP1559Transaction extends BaseTransaction<FeeMark
   }
 
   /**
-   * Returns the serialized encoding of the transaction.
+   * Returns the serialized encoding of the EIP-1559 transaction.
+   *
+   * Format: `0x02 || rlp([chainId, nonce, maxPriorityFeePerGas, maxFeePerGas, gasLimit, to, value, data,
+   * accessList, signatureYParity, signatureR, signatureS])`
+   *
+   * Note that in contrast to the legacy tx serialization format this in not
```
Comment (ryanio) [inline packages/tx/src/eip1559Transaction.ts:268]:
```suggestion
   * Note that in contrast to the legacy tx serialization format this is not
```

File: packages/tx/src/eip1559Transaction.ts [hunk 3]
```diff
@@ -234,15 +260,30 @@ export default class FeeMarketEIP1559Transaction extends BaseTransaction<FeeMark
...
   /**
-   * Returns the serialized unsigned tx (hashed or raw), which is used to sign the transaction.
+   * Returns the serialized unsigned tx (hashed or raw), which can be used
+   * to sign the transaction (e.g. for sending to a hardware wallet).
+   *
+   * Note: in contrast to the legacy tx the raw message format is already
+   * serialized and doesn't need to be RLP encoded any more.
+   *
+   * ```javascript
+   * const serializedMessage = tx.getMessageToSign(false) // use this for the HW wallet input
```
Comment (ryanio) [inline packages/tx/src/eip1559Transaction.ts:285]: 👍

File: packages/tx/src/eip2930Transaction.ts [hunk 4]
```diff
@@ -213,15 +230,30 @@ export default class AccessListEIP2930Transaction extends BaseTransaction<Access
   }
 
   /**
-   * Returns the serialized encoding of the transaction.
+   * Returns the serialized encoding of the EIP-2930 transaction.
+   *
+   * Format: `0x01 || rlp([chainId, nonce, gasPrice, gasLimit, to, value, data, accessList,
+   * signatureYParity (v), signatureR (r), signatureS (s)])`
+   *
+   * Note that in contrast to the legacy tx serialization format this in not
```
Comment (ryanio) [inline packages/tx/src/eip2930Transaction.ts:238]:
```suggestion
   * Note that in contrast to the legacy tx serialization format this is not
```

**Comment by ryanio** [comment]
overall looks great!

File: packages/tx/src/baseTransaction.ts (hunk 1 above)
Comment (holgerd77) [reply packages/tx/src/baseTransaction.ts:166]: Yes, it's actually written in the comments. 😄

File: packages/tx/src/eip1559Transaction.ts [hunk 5]
```diff
@@ -23,33 +29,50 @@ export default class FeeMarketEIP1559Transaction extends BaseTransaction<FeeMark
...
+   */
   public static fromTxData(txData: FeeMarketEIP1559TxData, opts: TxOptions = {}) {
     return new FeeMarketEIP1559Transaction(txData, opts)
   }
 
   /**
    * Instantiate a transaction from the serialized tx.
    *
-   * Note: this means that the Buffer should start with 0x01.
+   * Format: `0x02 || rlp([chainId, nonce, maxPriorityFeePerGas, maxFeePerGas, gasLimit, to, value, data,
```
[the remark this answers is not in the document]
Comment (holgerd77) [reply packages/tx/src/eip1559Transaction.ts:74]: @ryanio could you eventually do a short commit with the changes requested and then directly merge?

[the remark this answers is not in the document]
Comment (holgerd77) [reply packages/tx/src/eip1559Transaction.ts:74]: Not sure what to apply here instead eventually, maybe you have got a good idea. (`concat(..., ...)`?)

[the remark this answers is not in the document]
Comment (holgerd77) [reply packages/tx/src/eip1559Transaction.ts:74]: But I would say maybe it's also ok? But if you have got a better idea feel free to apply.

[the remark this answers is not in the document]
Comment (ryanio) [reply packages/tx/src/eip1559Transaction.ts:74]: hmmm, let's just keep it to the spec notation for now, changing it to anything else might be even more confusing.

we could add this small definition somewhere, but no urgency for now, I don't necessarily want to add it to every typedoc that references it:

> `||` is the byte/byte-array concatenation operator.

**Review by ryanio — APPROVED** [review]
lgtm!

**Pull request merged**